### PR TITLE
Add stricter permissions option to file permissions template

### LINKED ISSUE #6431: [RFE] Add option to check stricter file permissions on file_permissions template
Currently the file_permissions template only support checking a for specific file permissions value. It's not possible to check for stricter permissions, which in some cases is what security requirements look for:

- The OVAL template
  - https://github.com/ComplianceAsCode/content/blob/739c3635406e983d48dfe0e6d5dac8c634180151/shared/templates/file_permissions/oval.template#L22
- The input data processing
  - https://github.com/ComplianceAsCode/content/blob/739c3635406e983d48dfe0e6d5dac8c634180151/shared/templates/file_permissions/template.py#L12

#### Example on how to check for stricter permissions:
https://github.com/ComplianceAsCode/content/blob/739c3635406e983d48dfe0e6d5dac8c634180151/linux_os/guide/system/auditing/auditd_configure_rules/file_permissions_var_log_audit/oval/shared.xml#L36

Notice the `negate=true` which should do the trick to check for stricter permissions, it checks if there is any occurrence that would violate the permission. In case it's a stricter permission, it won't match any of those permission bits and the check pass if it's stricter.

### PR DESCRIPTION
**Pull request by ggbecker**
#### Description:

- Add stricter permissions option to file permissions template

#### Rationale:

- OVAL check will pass even if the permissions are more restrictive.

- Fixes #6431

### CODE REVIEW

**Comment by JAORMX** [comment]
worked for me. Let's wait for CI before merging.

**Comment by JAORMX** [comment]
LGTM

File: applications/openshift/master/file_permissions_multus_conf/rule.yml
Comment (ggbecker) [reply applications/openshift/master/file_permissions_multus_conf/rule.yml:36]: I was thinking about the naming and that might be a good idea.

File: docs/manual/developer/06_contributing_with_content.md
Comment (ggbecker) [reply docs/manual/developer/06_contributing_with_content.md:1343]: agreed. Thanks.

**Review by yuumasato — APPROVED** [review]

**Comment by yuumasato** [comment]
This is very nice improvement for the template, thanks @ggbecker ! :)

**Pull request merged**
